apps/extension/src/wasm-build-action.ts

```diff
@@ -38,6 +39,10 @@ const workerListener = ({ data }: { data: ActionBuildRequest }) => {
 
 self.addEventListener('message', workerListener, { once: true });
 
+const actionTypeRequiresProvingKey = (
+  actionPlanType: Exclude<TransactionPlan['actions'][number]['action']['case'], undefined>,
```

Comment:
Think this is a good candidate to extract:

```typescript
type ActionPlan = Exclude<TransactionPlan['actions'][number]['action']['case'], undefined>;

const actionTypeRequiresProvingKey = (actionPlanType: ActionPlan) =>
```

Reply:
Good call, done